### ISSUE TITLE
Refactor Travis Scanner method scan.py

### ISSUE THREAD

**Issue by michalslowikowski00** [apache/kibble-1#97]
URL checker could be split into smaller methods. It's hard to read a little bit right now.
This could help to write tests for the URL checker.
Code: [here is the code](https://github.com/apache/kibble/blob/639f10c260dcb3d9f2fa1098d4ee0ca30ecc9aeb/kibble/scanners/scanners/travis.py#L220
)

**Description**
Split  `scan.py` method for smaller chunks.

**Use case**
- refactor code 
- simplify the logic
- add tests

**Related Issues**
N/A

**Comment by turbaszek** [comment]
We should probably introduce something like `BaseSensor` class and refactor all scanners

**Comment by michalslowikowski00** [comment]
Yes, I was thinking about it. I need to check and understand how scanners work and find common patterns or smth like that. The first thing that came to my mind is the same pattern as we have in Airflow -- the base operator and base hook.
